app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -12,36 +12,58 @@ import mattermostManaged from 'app/mattermost_managed';
 
 import Config from 'assets/config';
 
+import {escapeRegex} from 'app/utils/markdown';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
 import {normalizeProtocol} from 'app/utils/url';
 
 export default class MarkdownLink extends PureComponent {
     static propTypes = {
+        actions: PropTypes.shape({
+            loadChannelsByTeamName: PropTypes.func.isRequired
+        }).isRequired,
         children: CustomPropTypes.Children.isRequired,
         href: PropTypes.string.isRequired,
-        onLongPress: PropTypes.func
+        onLongPress: PropTypes.func,
+        onPermalinkPress: PropTypes.func,
+        serverURL: PropTypes.string.isRequired,
+        siteURL: PropTypes.string.isRequired
     };
 
     static defaultProps = {
-        onLongPress: () => true
+        onLongPress: () => true,
+        onPermalinkPress: () => true
     };
 
     static contextTypes = {
         intl: intlShape.isRequired
     };
 
-    handlePress = () => {
-        const url = normalizeProtocol(this.props.href);
+    handlePress = wrapWithPreventDoubleTap(() => {
+        const {actions, href, onPermalinkPress, serverURL, siteURL} = this.props;
+        const url = normalizeProtocol(href);
 
         if (!url) {
             return;
         }
 
-        Linking.canOpenURL(url).then((supported) => {
-            if (supported) {
-                Linking.openURL(url);
-            }
-        });
-    };
+        const pattern = new RegExp('^' + escapeRegex(serverURL) + '\\/([^\\/]+)\\/pl\\/(\\w+)');
+        const sitePattern = new RegExp('^' + escapeRegex(siteURL) + '\\/([^\\/]+)\\/pl\\/(\\w+)');
```

Comment:
Very minor and is non-blocking: just thinking if it makes sense to have the entire line be part of `escapeRegex` function?
```
new RegExp('^' + escapeRegex(siteURL) + '\\/([^\\/]+)\\/pl\\/(\\w+)');
``

Reply:
The `escapeRegex` function only matters for the serverURL and siteURL since those are user defined, and having characters like `*` or `[` in them would cause the `new RegExp` to error out. The rest of the pattern is already escaped.